### PR TITLE
Add OpenShift cleaner script

### PR DESCRIPTION
**Pull request by atodorov**

### CODE REVIEW

**Comment by atodorov** [comment]
Example of running this script locally:

```
WARNING: Using insecure TLS client config. Setting this option is not supported!

Logged into "https://api.ocp-virt.prod.psi.redhat.com:6443" as "system:serviceaccount:image-builder:gitab-ci" using the token provided.

You have access to the following projects and can switch between them with 'oc project <projectname>':

    cnv-qe
  * image-builder
    openshift-virtualization-os-images

Using project "image-builder".

'ATTENTION: Feb 6 2023, the my_ldap_provider will no longer be available
as a login option. Please see the "[PSI Openshift] Important Authentication Update"
email for details.'
system:serviceaccount:image-builder:gitab-ci
Already on project "image-builder" on server "https://api.ocp-virt.prod.psi.redhat.com:6443".
virtualmachine.kubevirt.io "image-builder-vm-rhel-8-10-x86-64-main-6311253374" deleted
The VM 'image-builder-vm-rhel-8-10-x86-64-main-6311253374' was terminated
persistentvolumeclaim "image-builder-data-volume-rhel-8-10-x86-64-main-6311253374" deleted
The PVC 'image-builder-data-volume-rhel-8-10-x86-64-main-6311253374' was terminated
persistentvolumeclaim "image-builder-data-volume-rhel-8-10-x86-64-pr-4032-6449247316" deleted
The PVC 'image-builder-data-volume-rhel-8-10-x86-64-pr-4032-6449247316' was terminated
persistentvolumeclaim "image-builder-data-volume-rhel-8-10-x86-64-pr-4032-6449247316-scratch" deleted
The PVC 'image-builder-data-volume-rhel-8-10-x86-64-pr-4032-6449247316-scratch' was terminated
persistentvolumeclaim "image-builder-data-volume-rhel-8-10-x86-64-pr-4032-6450318042" deleted
The PVC 'image-builder-data-volume-rhel-8-10-x86-64-pr-4032-6450318042' was terminated
```

File: openshift.sh [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+# unescaped newlines which break echo
+ALL_VMS=$($OC_CLI get vm --output json | jq 'del(.items[].spec.template)' | jq 'del(.items[].status.volumeSnapshotStatuses)' | jq -c '.items[]')
+for INSTANCE in ${ALL_VMS}; do
+    REMOVE=1
+    VM_NAME=$(echo "${INSTANCE}" | jq -r '.metadata.name')
+    CREATION_TIME=$(echo "${INSTANCE}" | jq -r '.metadata.creationTimestamp')
+
+    if [[ $(date -d "${CREATION_TIME}" +%s) -gt "${DELETE_TIME}" ]]; then
+        REMOVE=0
+        echo "The VM '${VM_NAME}' was launched less than ${HOURS_BACK} hours ago"
```
Comment (atodorov) [inline openshift.sh:44]: Currently this is 6 hours and I think we'll need to drastically reduce it b/c resource quota in OpenShift is very low. 

Probably reduce down to 1hr? WDYT ?

**Comment by atodorov** [comment]
> The PVC 'image-builder-data-volume-rhel-8-10-x86-64-main-6311253374' was terminated
> persistentvolumeclaim "image-builder-data-volume-rhel-8-10-x86-64-pr-4032-6449247316" deleted

Hmmm, even though this is supposed to be removed I can still see the PVC in the OpenShift UI. 

image

executing the removal script again this time tells me that this PVC was created recently and will not be removed:

> PVC 'image-builder-data-volume-rhel-8-10-x86-64-pr-4032-6449247316' was created less than 6 hours ago


Not quite sure what's going on here.

File: openshift.sh (hunk 1 above)
Comment (jrusz) [reply openshift.sh:44]: Sure, I don't see an issue with that. I don't even know why we have it set so high, any of these test cloud instances should be alive less than 10 minutes, if it's longer there is something wrong...

File: openshift.sh [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    fi
+
+    # TODO: check for tag persist=true
```
Comment (jrusz) [inline openshift.sh:47]: Got a ticket for this or do you plan to add it in this PR?

**Comment by jrusz** [comment]
> > The PVC 'image-builder-data-volume-rhel-8-10-x86-64-main-6311253374' was terminated
> > persistentvolumeclaim "image-builder-data-volume-rhel-8-10-x86-64-pr-4032-6449247316" deleted
> 
> Hmmm, even though this is supposed to be removed I can still see the PVC in the OpenShift UI.
> 
> image
> 
> executing the removal script again this time tells me that this PVC was created recently and will not be removed:
> 
> > PVC 'image-builder-data-volume-rhel-8-10-x86-64-pr-4032-6449247316' was created less than 6 hours ago
> 
> Not quite sure what's going on here.

Hm, that's strange. Can you delete it manually? I think that once you delete the VM you should be able to delete the PVC... Maybe a slight delay would be required between the two?

**Review by jrusz — DISMISSED** [review]
LGTM, just one question and I'm not sure what's up with that PVC, seems more investigation is required.

File: openshift.sh (hunk 2 above)
Comment (atodorov) [reply openshift.sh:52]: I was not planning to add the tag functionality as part of this PR. Documented as https://issues.redhat.com/browse/COMPOSER-2195

**Comment by atodorov** [comment]
FTR I am having issues deleting some of the PVCs, created RITM1742240 ticket to ask for help.

**Pull request merged**